### PR TITLE
diesel-cli: 1.4.0 -> 1.4.1

### PR DESCRIPTION
**Pull request by ivan**
###### Motivation for this change

An update was requested in #117899

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@winterqt and other diesel-cli users, please test and comment

### CODE REVIEW

**Comment by ivan** [comment]
I did not add barrel support because of https://github.com/NixOS/nixpkgs/issues/117899#issuecomment-809534794, but if it is possible to build barrel for all backends, please let me know.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117956` at 44479cef run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>diesel-cli</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/tools/diesel-cli/default.nix:79:5:
```
   |
79 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`checkPhase` should probably contain `runHook preCheck` and `runHook postCheck`.

Near pkgs/development/tools/diesel-cli/default.nix:62:3:
```
   |
62 |   checkPhase = optionalString sqliteSupport ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/diesel-cli/fix-deprecated.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+diff --git a/diesel/src/query_builder/insert_statement/mod.rs b/diesel/src/query_builder/insert_statement/mod.rs
+index b6bb34df9d..7486eb7a59 100644
+--- a/diesel/src/query_builder/insert_statement/mod.rs
++++ b/diesel/src/query_builder/insert_statement/mod.rs
+@@ -1,3 +1,5 @@
++#![allow(useless_deprecated)]
```
Comment (Mic92) [inline pkgs/development/tools/diesel-cli/fix-deprecated.patch:6]: Should we try to upstream this patch as well?

**Comment by Mic92** [comment]
Result of `nixpkgs-review pr 117956` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>diesel-cli</li>
  </ul>
</details>

**Pull request merged**